tests/integration/backup/self_reference/import_test.go

```diff
@@ -0,0 +1,365 @@
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Author {
+						name: String
+						book: Book @relation(name: "author_book")
+						reviewed: Book @relation(name: "reviewedBy_reviewed")
+					}
+					type Book {
+						name: String
+						author: Author @primary @relation(name: "author_book")
+						reviewedBy: Author @primary @relation(name: "reviewedBy_reviewed")
+					}
+				`,
+			},
+			testUtils.BackupImport{
+				ImportContent: `{
+					"Author":[
+						{
+							"name":"John"
+						}
+					],
+					"Book":[
+						{
+							"name":"John and the sourcerers' stone",
+							"author":"bae-decf6467-4c7c-50d7-b09d-0a7097ef6bad",
+							"reviewedBy":"bae-decf6467-4c7c-50d7-b09d-0a7097ef6bad"
+						}
+					]
+				}`,
+			},
+			testUtils.Request{
+				Request: `
+					query {
+						Book {
+							name
+							author {
+								name
+								reviewed {
+									name
+								}
+							}
+						}
+					}`,
+				Results: []map[string]any{
+					{
+						"name": "John and the sourcerers' stone",
+						"author": map[string]any{
+							"name": "John",
+							"reviewed": map[string]any{
+								"name": "John and the sourcerers' stone",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, []string{"Author", "Book"}, test)
+}
+
+func TestBackupSelfRefImport_PrimaryRelationWithSecondCollectionWrongOrder_NoError(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Author {
+						name: String
+						book: Book @relation(name: "author_book")
+						reviewed: Book @relation(name: "reviewedBy_reviewed")
+					}
+					type Book {
+						name: String
+						author: Author @primary @relation(name: "author_book")
+						reviewedBy: Author @primary @relation(name: "reviewedBy_reviewed")
+					}
+				`,
+			},
+			testUtils.BackupImport{
+				ImportContent: `{
+					"Book":[
+						{
+							"name":"John and the sourcerers' stone",
+							"author":"bae-decf6467-4c7c-50d7-b09d-0a7097ef6bad",
+							"reviewedBy":"bae-decf6467-4c7c-50d7-b09d-0a7097ef6bad"
+						}
+					],
+					"Author":[
+						{
+							"name":"John"
+						}
+					]
+				}`,
+			},
+			testUtils.Request{
+				Request: `
+					query {
+						Book {
+							name
+							author {
+								name
+								reviewed {
+									name
+								}
+							}
+						}
+					}`,
+				Results: []map[string]any{
+					{
+						"name": "John and the sourcerers' stone",
+						"author": map[string]any{
+							"name": "John",
+							"reviewed": map[string]any{
+								"name": "John and the sourcerers' stone",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, []string{"Author", "Book"}, test)
+}
+
+// This test documents undesirable behaviour, as the documents are not linked.
+// https://github.com/sourcenetwork/defradb/issues/1697
```

Comment:
note: This test isn't linked to an import error. It's an export error. It's having trouble setting the right expected book ID because they both have a primary field related to each other.

Reply:
Ah nice spot - would be good to fix that too if you have time, I'm not sure your fix-PR solves for this kind of schema anyway and it is hard to test fully independently of the import

Reply:
We can't test it on import. It has to be tested on export. The json object representing the exported data is clearly wrong so we could correct it manually and assert an error instead, Showing a wrong output after import is fine. Just a little less clear as to what the problem is.

Reply:
Ah true - good shout.  Tweaking now, will like to a new ticket